### PR TITLE
SQL examples draft

### PR DESCRIPTION
**Pull request by pdobacz**
Part of #77. I didn't test these, nor I don't know how do they present.

Edon: Feel free to just plug this into your scaffolding, and once this is somewhat working, I can pick up again and start polishing the looks and rules.

### CODE REVIEW

File: src/main/metabase/examples.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+function lines(...lines: string[]) {
+  return lines.join('\n');
+}
+
+const numberFieldTypes = ['int2', 'int4', 'int8', 'float4', 'float8', 'numeric'];
+
+function rawGroupBySQL(column: string, table: string, displayName: string): ExampleInfo {
+  return {
+    name: `${displayName} by ${column}`,
+    sql: lines(`SELECT ${postgresQuote(column)}`, `FROM ${postgresQuote(table)}`),
```
Comment (edongashi) [inline src/main/metabase/examples.ts:39]: Is GROUP BY clause missing?

File: src/main/metabase/examples.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  };
+}
+
+function countDistinctSQL(column: string, table: string): ExampleInfo {
+  return {
+    name: `Distinct ${column}`,
+    sql: lines(`SELECT count(distinct ${postgresQuote(column)}) as distinct_${column}`, `FROM ${postgresQuote(table)}`),
```
Comment (edongashi) [inline src/main/metabase/examples.ts:46]: Do we handle quoting of this? If so, we might get `distinct_"some col"`

File: src/main/metabase/examples.ts (hunk 1 above)
Comment (pdobacz) [reply src/main/metabase/examples.ts:39]: yup, lost in refactors

File: src/main/metabase/examples.ts (hunk 2 above)
Comment (pdobacz) [reply src/main/metabase/examples.ts:46]: I think I checked mb reports the table unquoted here, so we should be fine. I'll check again

Comment (pdobacz) [reply src/main/metabase/examples.ts:46]: unquoted, but I have to prevent `distinct_some col`, I'll fix this

Comment (edongashi) [reply src/main/metabase/examples.ts:46]: > I'll fix this

Let's fix bugs in a future PR. I want to synchronize and experiment often.

**Review by edongashi — DISMISSED** [review]

File: src/main/metabase/examples.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    sql: lines(`SELECT ${bucket} as ${column}_year, count(*)`, `FROM ${postgresQuote(table)}`, `GROUP BY ${bucket}`),
+  };
+}
+
+function makeExampleInfos(field: Field, table: Table, aidColumns: string[]): ExampleInfo[] {
+  try {
+    if (field.semantic_type === 'type/PK' || field.database_type === 'serial') {
+      // No sensible example for columns being just row IDs.
+      return [];
+    } else if (aidColumns.includes(field.name)) {
```
Comment (edongashi) [inline src/main/metabase/examples.ts:82]: I'd split (in a future revision) all these into their own function, i.e. `aidColumnExamples`, `numericColumnExamples`, `textColumnExamples`, etc.

Also a future idea is to get some columns with very few distinct values and try grouping by 2 columns. If the values are distributed enough we might get non-suppressed results.

**Review by edongashi — APPROVED** [review]

**Pull request merged**